### DISCUSSION TITLE
Could not start training, please help, thank you!

### DISCUSSION THREAD

**Discussion by 764220145** [d8ahazard/sd_dreambooth_extension#1050] [Q&A] [answered]
when I click traning, the output will show "Exception training model: 'import of xformers halted; None in sys.modules"
and then the training is stop. I do not know why, and here is the codes.
If you have any idea to solve this, pls help me. Thank you!

Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Custom model name is Luotianyi_1stTime
Initializing dreambooth training...
The version of diffusers is less than or equal to 0.14.0. Performing monkey-patch...
Pre-processing images: LTY: : 0it [00:00, ?it/s]
Nothing to generate.
Traceback (most recent call last):
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\ui_functions.py", line 681, in start_training
    result = main(use_txt2img=use_txt2img)
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 1148, in main
    return inner_loop()
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\memory.py", line 123, in decorator
    return function(batch_size, grad_size, prof, *args, **kwargs)
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 261, in inner_loop
    import xformers
ModuleNotFoundError: import of xformers halted; None in sys.modules
Restored system models.

**Comment by Arilziem** [comment, marked as answer]
Can you post the version info that comes up when you start?

```text
Python revision: xy
Dreambooth revision: xy
SD-WebUI revision: xy

Checking Dreambooth requirements...
[+] torch version ....
```

and so on

**Reply by 764220145** [reply]
Sorry about that, here is the version info, and thank for your reply.

venv "AIDrawing\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: 0cc0ee1bcb4c24a8c9715f66cede06601bfc00c8
Installing requirements for Web UI

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Dreambooth revision: bff61c6b92b79ece2e140ee240e0628dd9a4ebef
SD-WebUI revision: 0cc0ee1bcb4c24a8c9715f66cede06601bfc00c8

Checking Dreambooth requirements...
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64"' don't match your environment
Ignoring mediapipe-silicon: markers 'sys_platform == "darwin"' don't match your environment
Collecting accelerate==0.16.0
  Using cached accelerate-0.16.0-py3-none-any.whl (199 kB)
Collecting gitpython~=3.1.31
  Using cached GitPython-3.1.31-py3-none-any.whl (184 kB)
Collecting transformers~=4.26.1
  Using cached transformers-4.26.1-py3-none-any.whl (6.3 MB)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Installing collected packages: requests, gitpython, accelerate, transformers
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Uninstalling requests-2.25.1:
      Successfully uninstalled requests-2.25.1
  Attempting uninstall: gitpython
    Found existing installation: GitPython 3.1.27
    Uninstalling GitPython-3.1.27:
      Successfully uninstalled GitPython-3.1.27
  Attempting uninstall: accelerate
    Found existing installation: accelerate 0.12.0
    Uninstalling accelerate-0.12.0:
      Successfully uninstalled accelerate-0.12.0
  Attempting uninstall: transformers
    Found existing installation: transformers 4.25.1
    Uninstalling transformers-4.25.1:
      Successfully uninstalled transformers-4.25.1
Successfully installed accelerate-0.16.0 gitpython-3.1.31 requests-2.28.2 transformers-4.26.1

[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
[+] accelerate version 0.16.0 installed.
[+] bitsandbytes version 0.35.4 installed.
[+] diffusers version 0.13.1 installed.
[+] transformers version 4.26.1 installed.
[+] xformers version 0.0.17.dev464 installed.

#######################################################################################################

Launching Web UI with arguments:
No module 'xformers'. Proceeding without it.
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
Script path is D:\Stable Diffusion\stable-diffusion-webui
Loading weights [a074b8864e] from D:\Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\Counterfeit-V2.5.safetensors
Creating model from config: D:\Stable Diffusion\stable-diffusion-webui\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying cross attention optimization (Doggettx).
Textual inversion embeddings loaded(1): luotianyi
Model loaded in 10.4s (load weights from disk: 0.2s, create model: 0.5s, apply weights to model: 6.2s, apply half(): 0.6s, move model to device: 0.8s, load textual inversion embeddings: 2.1s).

**Reply by 764220145** [reply]
I added prompt text for each picture and it did generating images for training..., but still have the same issue. 
Here is the code
venv "AIDrawing\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: 0cc0ee1bcb4c24a8c9715f66cede06601bfc00c8
Installing requirements for Web UI

#######################################################################################################
Initializing Dreambooth
If submitting an issue on github, please provide the below text for debugging purposes:

Python revision: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Dreambooth revision: bff61c6b92b79ece2e140ee240e0628dd9a4ebef
SD-WebUI revision: 0cc0ee1bcb4c24a8c9715f66cede06601bfc00c8

Checking Dreambooth requirements...
Ignoring tensorflow-macos: markers 'sys_platform == "darwin" and platform_machine == "arm64"' don't match your environment
Ignoring mediapipe-silicon: markers 'sys_platform == "darwin"' don't match your environment
Collecting accelerate==0.16.0
  Using cached accelerate-0.16.0-py3-none-any.whl (199 kB)
Collecting gitpython~=3.1.31
  Using cached GitPython-3.1.31-py3-none-any.whl (184 kB)
Collecting transformers~=4.26.1
  Using cached transformers-4.26.1-py3-none-any.whl (6.3 MB)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Installing collected packages: requests, gitpython, accelerate, transformers
  Attempting uninstall: requests
    Found existing installation: requests 2.25.1
    Uninstalling requests-2.25.1:
      Successfully uninstalled requests-2.25.1
  Attempting uninstall: gitpython
    Found existing installation: GitPython 3.1.27
    Uninstalling GitPython-3.1.27:
      Successfully uninstalled GitPython-3.1.27
  Attempting uninstall: accelerate
    Found existing installation: accelerate 0.12.0
    Uninstalling accelerate-0.12.0:
      Successfully uninstalled accelerate-0.12.0
  Attempting uninstall: transformers
    Found existing installation: transformers 4.25.1
    Uninstalling transformers-4.25.1:
      Successfully uninstalled transformers-4.25.1
Successfully installed accelerate-0.16.0 gitpython-3.1.31 requests-2.28.2 transformers-4.26.1

[+] torch version 1.13.1+cu117 installed.
[+] torchvision version 0.14.1+cu117 installed.
[+] accelerate version 0.16.0 installed.
[+] bitsandbytes version 0.35.4 installed.
[+] diffusers version 0.13.1 installed.
[+] transformers version 4.26.1 installed.
[+] xformers version 0.0.17.dev464 installed.

#######################################################################################################

Launching Web UI with arguments:
No module 'xformers'. Proceeding without it.
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
Script path is D:\Stable Diffusion\stable-diffusion-webui
Loading weights [a074b8864e] from D:\Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\Counterfeit-V2.5.safetensors
Creating model from config: D:\Stable Diffusion\stable-diffusion-webui\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying cross attention optimization (Doggettx).
Textual inversion embeddings loaded(1): luotianyi
Model loaded in 4.0s (create model: 0.4s, apply weights to model: 0.9s, apply half(): 0.8s, move model to device: 0.8s, load textual inversion embeddings: 1.1s).
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:22<00:00,  1.10s/it]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:21<00:00,  1.07s/it]
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:17<00:00,  1.15it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:19<00:00,  1.05it/s]
Loading weights [89d59c3dde] from D:\Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\animefull.ckpt
Applying cross attention optimization (Doggettx).
Weights loaded in 2.9s (load weights from disk: 1.8s, apply weights to model: 0.4s, move model to device: 0.7s).
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:16<00:00,  1.22it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:18<00:00,  1.11it/s]
Loading weights [a074b8864e] from D:\Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\Counterfeit-V2.5.safetensors
Applying cross attention optimization (Doggettx).
Weights loaded in 1.3s (apply weights to model: 0.7s, move model to device: 0.6s).
Couldn't find Lora with name IPV1
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.47it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:15<00:00,  1.30it/s]
Loading weights [89d59c3dde] from D:\Stable Diffusion\stable-diffusion-webui\models\Stable-diffusion\animefull.ckpt
Applying cross attention optimization (Doggettx).
Weights loaded in 2.8s (load weights from disk: 1.6s, apply weights to model: 0.4s, move model to device: 0.8s).
Couldn't find Lora with name IPV1
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.50it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:15<00:00,  1.32it/s]
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Wizard results:<br>Num Epochs: 150<br>Num instance images per class image: 5
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Model dir set to: D:\Stable Diffusion\stable-diffusion-webui\models\dreambooth\Luotianyi_FirstTime
Custom model name is Luotianyi_1stTime
Initializing dreambooth training...
The version of diffusers is less than or equal to 0.14.0. Performing monkey-patch...
Pre-processing images: 100_LuoTianYi: : 0it [00:00, ?it/s]                                                             We need a total of 350 class images.
Pre-processing images: 100_LuoTianYi: : 0it [00:00, ?it/s]
Generating 350 class images for training...
Generating class images 0/350::   0%|                                                          | 0/350 [00:00<?, ?it/s]Using scheduler: DDIM
100%|██████████████████████████████████████████████████████████████████████████████████| 10/10 [00:02<00:00,  3.90it/s]
a lot ██████████████████████████████████████████████████████████████████████████████
100%|██████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 10.77it/s]
Generating class images 350/350:: 100%|██████████████████████████████████████████| 350/350 [00:00<00:00, 349775.17it/s]Restored system models.
Generated 350 new class images.
Generating class images 350/350:: 100%|█████████████████████████████████████████████| 350/350 [00:00<00:00, 425.23it/s]
**Traceback (most recent call last):
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\ui_functions.py", line 681, in start_training
    result = main(use_txt2img=use_txt2img)
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 1148, in main
    return inner_loop()
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\memory.py", line 123, in decorator
    return function(batch_size, grad_size, prof, *args, **kwargs)
  File "D:\Stable Diffusion\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 261, in inner_loop
    import xformers
ModuleNotFoundError: import of xformers halted; None in sys.modules
Restored system models.**

**Reply by Arilziem** [reply]
Try adding `--xformers` to your launch arguments in webui-user-dreambooth.bat, that might be it

**Reply by 764220145** [reply]
I think it works, it has processes now!
Steps:   1%|       | 176/28000 [01:24<3:32:18,  2.18it/s, inst_loss=0, loss=0.481, lr=1e-5, prior_loss=0.641, vram=9.1]
Thank you so much!
